internal/k8s/nodes/nodes.go

```diff
@@ -3,9 +3,35 @@
 package nodes
 
 import (
+	"strings"
+
 	corev1 "k8s.io/api/core/v1"
 )
 
+func IsNodeAvailable(n *corev1.Node) (bool, string) {
+	ret := true
+	reason := []string{}
+
+	if IsNetworkUnavailable(n) {
+		ret = ret && false
```

Comment:
We can exit early with the first offending reason. Also, I am tempted to say that (bool, string) is worse than returning a speaking error.

Reply:
speaking error then, not great because this is not really an error but I agree (bool,string) or (string) and compare to empty is worse.